### PR TITLE
Add documentation for compatibility versions

### PR DESCRIPTION
**Pull request by howardjohn**
DNM until https://github.com/istio/istio/pull/48449 merges

### CODE REVIEW

**Review by linsun — COMMENTED** [review]
Doc looks straightforward.

Should we add this is experimental first?

File: content/en/docs/setup/additional-setup/compatibility-versions/index.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Using compatibility versions
+
+To use a compatibility version, simple set the `compatibilityVersion` field.
+
+{{< tabset category-name="install" >}}
+{{< tab name="IstioOperator" category-value="iop" >}}
+
+{{< text shell >}}
+$ helm install ... --set compatibilityVersion={{< istio_previous_version >}}
```
Comment (linsun) [inline content/en/docs/setup/additional-setup/compatibility-versions/index.md:28]: wrong cmd - expect istioctl here?

File: content/en/docs/setup/additional-setup/compatibility-versions/index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+{{< /text >}}
+
+{{< /tab >}}
+{{< tab name="Helm" category-value="helm" >}}
+
+{{< text shell >}}
+$ istioctl install --set values.compatibilityVersion={{< istio_previous_version >}}
```
Comment (linsun) [inline content/en/docs/setup/additional-setup/compatibility-versions/index.md:35]: except helm cmd here?

File: content/en/docs/setup/additional-setup/compatibility-versions/index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+description: How to configure "compatibility versions", to decouple behavioral changes from releases.
+weight: 36
+keywords: [profiles,install,helm]
+owner: istio/wg-environments-maintainers
+test: n/a
+---
+
+With each new version of Istio, there may be some intentional behavioral changes.
+These can be to improve security, fix incorrect behavior, or otherwise improve Istio for users.
+Generally, these types of changes impact only edge cases
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/compatibility-versions/index.md:12]:
```suggestion
Generally, these types of changes impact only edge cases.
```

File: content/en/docs/setup/additional-setup/compatibility-versions/index.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+While beneficial on the long term, each behavioral change introduces risk for during upgrades.
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/compatibility-versions/index.md:14]: 'for' or 'during' or add a '/'.

File: content/en/docs/setup/additional-setup/compatibility-versions/index.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+$ istioctl install --set values.compatibilityVersion={{< istio_previous_version >}}
+{{< /text >}}
+
+{{< /tab >}}
+{{< /tabset >}}
+
+## When should I use compatibility versions?
+
+Compatibility versions are recommended to be used only when an incompatibility is found, rather than as the default.
+Each compatibility version will only persist for a few releases, so eventually you will need to migrate to the new behavior.
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/compatibility-versions/index.md:44]: Wondering if we need a general rule how long these should persist? Deprecation rule?

**Comment by howardjohn** [comment]
> Doc looks straightforward.
> 
> Should we add this is experimental first?

I don't know how we can make this "experimental" from a technical standpoint. Its really just an alias for `profile` which have existed for years

**Review by dhawton — COMMENTED** [review]
One small typo is all I see... otherwise LGTM

**Review by kfaseela — APPROVED** [review]

**Comment by kfaseela** [comment]
LGTM

**Pull request merged**